core/src/main/scala/org/apache/spark/sql/TiStrategy.scala

```diff
@@ -308,8 +308,13 @@ case class TiStrategy(getOrCreateTiContext: SparkSession => TiContext)(sparkSess
     sortOrder: Seq[SortOrder]
   ): SparkPlan = {
     val request = new TiDAGRequest(pushDownType(), timeZoneOffsetInSeconds())
-    request.setLimit(limit)
-    addSortOrder(request, sortOrder)
+    val filteredSortOrder =
+      sortOrder.filter(order => TiUtil.isSupportedOrderBy(order, source, blacklist))
+    if (filteredSortOrder.nonEmpty) {
```

Comment:
does this mean that limit will not be pushed down if there isn't an `order by` statement? It seems quite strange.

Reply:
yes.